tools/test/examples/README.md

```diff
@@ -0,0 +1,167 @@
+# Examples testing script
+
+The scripts in this folder are used for testing `mbed-os` official examples. It contains the following files:
+
+- `examples.py` - the main script that serves as the command-line interface.
+- `examples.json` - the default configuration file for the script, which contains the information to the examples. If required, you can pass a customized configuration file as an argument to the script.
```

Comment:
What do you mean by a customized configuration file? Can you show an example on how to use this feature? 

I think that we currently don't have a way to test different profiles (bare metal profile/minimal-printf profile) and/or library configurations, do we? Is this PR the first step towards achieving this?

Reply:
the  `json` template in the `README` is an example of configuration. This means if other people want to use this in their own CI, they can do it like such. But regarding to mbed-OS main CI, it always uses the default configuration file.

The new scripts actually allow the `minimal-printf` to be tested, you can use:

```
python mbed-os/tools/test/examples/examples.py compile GCC_ARM -m <target> --profile develop mbed-os/tools/profile/extensions/minimal-printf.json
```